examples/pos/namespaced_constructors.effekt

```diff
@@ -0,0 +1,20 @@
+type Foo {
+  Bar()
+}
+
+type Boo {
+  Bar()
+}
+type Bars {
+  Foo(f: Foo)
+  Boo(b: Boo)
+}
+
+def consume(b: Bars) = b match {
+  case Foo(Foo::Bar()) => 1
+  case Boo(Boo::Bar()) => 2
+}
```

Comment:
Here you can see that we now can also disambiguate equally named constructors in patterns.